### PR TITLE
wolfssl embOS port initialize

### PR DESCRIPTION
**Pull request by kaleb-himes**
Types - SIZEOF_LONG 4, SIZEOF_LONG_LONG 8

Endianness - Little Endian

NO_WRITEV

Input / Output: wolfCrypt only did not worry about I/O and socket API at this time

FileSystem - Not ported at this time, available through emFile https://www.segger.com/emfile.html

Threading - Will be similar to FreeRTOS, task specific and based on task priority Not ported at this time due to no threading needed in wolfcrypt only build

Random - available but without researching true randomness used the "Testing" random and defined NO_DEV_RANDOM for now

Memory - standard C malloc, free, realloc available, OS_malloc, free, realloc provided for thread safe if/when we port the Mutex operations.

Time - Not ported for wolfcrypt_test(), will port this if we decide to do benchmark

C Standard Library - available

Logging - printf works fine solved though the IDE and IAR-Jlink

wolfSSL available Public Key Operations - Tested and running well.

Public Key Operations (custom)- Not requested / specified (In reference to: Necessary if you want to use your own public key implementation from the porting guide.) 

Atomic Record Layer Processing - Not specified

Features - handled through user_settings.h

### CODE REVIEW

File: wolfssl/ssl.h [hunk 1]
```diff
@@ -969,7 +969,8 @@ WOLFSSL_API int wolfSSL_make_eap_keys(WOLFSSL*, void* key, unsigned int len,
             #include <sys/types.h>
             #include <sys/socket.h>
         #elif !defined(WOLFSSL_MDK_ARM) && !defined(WOLFSSL_IAR_ARM) && \
-              !defined(WOLFSSL_PICOTCP) && !defined(WOLFSSL_ROWLEY_ARM)
+              !defined(WOLFSSL_PICOTCP) && !defined(WOLFSSL_ROWLEY_ARM) && \
+              !defined(WOLFSSL_EMBOS)
```
Comment (toddouska) [inline wolfssl/ssl.h]: While it's not bad to port the TLS/SSL layer too that wasn't part of the crypt only scope.  Was crypt only adhered too?

File: wolfssl/wolfcrypt/settings.h [hunk 2]
```diff
@@ -409,6 +412,16 @@
     #endif
 #endif
 
+#ifdef WOLFSSL_EMBOS
+    #define USE_FAST_MATH
```
Comment (toddouska) [inline wolfssl/wolfcrypt/settings.h]: If the port is truly only settings then these should probably be example emboss settings.  What if someone doesn't want ecc?

File: wolfssl/wolfcrypt/settings.h [hunk 3]
```diff
@@ -409,6 +412,16 @@
...
+    #define HAVE_ECC
+    #define NO_DEV_RANDOM
```
Comment (toddouska) [inline wolfssl/wolfcrypt/settings.h]: How was the seed generated?

File: wolfssl/wolfcrypt/settings.h [hunk 4]
```diff
@@ -409,6 +412,16 @@
...
+    #define NO_FILESYSTEM           /* Not ported at this time */
+    #define USE_CERT_BUFFERS_1024   /* use when NO_FILESYSTEM */
```
Comment (toddouska) [inline wolfssl/wolfcrypt/settings.h]: We should prefer 2048 for security / testing.

File: wolfssl/ssl.h (hunk 1 above)
Comment (kaleb-himes) [inline wolfssl/ssl.h]: Yes crypt-only was adhered to. My apologies I had added this prior to that specification and was thinking "Future" decided not to remove. This was a judgment call.

File: IDE/IAR-EWARM/embOS/user_settings.h [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#ifndef _EMBOS_USER_SETTINGS_H_
+#define _EMBOS_USER_SETTINGS_H_
+
```
Comment (toddouska) [inline IDE/IAR-EWARM/embOS/user_settings.h]: How is this file going to be pulled in?  Was it tested?

Comment (kaleb-himes) [inline IDE/IAR-EWARM/embOS/user_settings.h]: Yes this was tested. It's pulled in through the IDE and is documented in the README I'm generating for the customer

File: wolfssl/wolfcrypt/settings.h (hunk 4 above)
Comment (kaleb-himes) [inline wolfssl/wolfcrypt/settings.h]: I can update this no problem.

File: wolfssl/wolfcrypt/settings.h (hunk 3 above)
Comment (kaleb-himes) [inline wolfssl/wolfcrypt/settings.h]: We used the #warning for testing purposes only generate seed for the sake of time. We can research this more.

File: wolfssl/wolfcrypt/settings.h (hunk 2 above)
Comment (kaleb-himes) [inline wolfssl/wolfcrypt/settings.h]: A valid point. Will move this to user_settings

File: IDE/IAR-EWARM/embOS/user_settings.h (hunk 5 above)
Comment (toddouska) [inline IDE/IAR-EWARM/embOS/user_settings.h]: The IDE files and README need to go into the pull request.  This isn't a one off but an addition that should be useful to others as well.

**Comment by toddouska** [comment]
Public key operations are definitely needed as part of wolfCrypt.  TLS is not needed.

File: wolfssl/wolfcrypt/settings.h (hunk 3 above)
Comment (toddouska) [inline wolfssl/wolfcrypt/settings.h]: I don't understand how a #warning would generate a seed.  You have defined NO_DEV_RANDOM which currently does:

#elif defined(NO_DEV_RANDOM)

#error "you need to write an os specific wc_GenerateSeed() here"

/*
int wc_GenerateSeed(OS_Seed* os, byte* output, word32 sz)
{
    return 0;
}
*/

The pull request doesn't reflect the seeding that I can tell.

**Comment by kaleb-himes** [comment]
Tasks for tomorrow:
Finish writing and add README
Add example_user_settings
remove HAVE_ECC from settings.h move to example_user_settings.h
research available random methods

**Pull request closed without merging**